mutt/charset.c

```diff
@@ -303,27 +304,24 @@ int mutt_ch_convert_nonmime_string(char **ps)
   if (!ps)
     return -1;
 
-  const char *c1 = NULL;
+  const struct ListNode *np = NULL;
 
-  for (const char *c = C_AssumedCharset; c; c = c1 ? c1 + 1 : 0)
+  STAILQ_FOREACH(np, &C_AssumedCharset->head, entries)
   {
+    const char *from = (const char *) np->data;
     char *u = *ps;
     size_t ulen = mutt_str_strlen(*ps);
 
     if (!u || !*u)
       return 0;
 
-    c1 = strchr(c, ':');
-    size_t n = c1 ? c1 - c : mutt_str_strlen(c);
+    size_t n = mutt_str_strlen(from);
     if (n == 0)
       return 0;
```

Comment:
We can simplify this.
The `DT_SLIST` stores empty strings as `NULL`.
(Currently, none of the config variables has `SLIST_ALLOW_EMPTY`)

I think we can reduce this to:
```c
if (!from)
  return 0;
```

and move it to the beginning of the `FOREACH`

Reply:
Looking more closely at this function, I spotted an error that I introduced -- the result of the conversion was being discarded.
I've pushed a fix and tidied the function.